nums/core/array/utils.py

```diff
@@ -129,6 +129,29 @@ def can_broadcast_shapes(a_shape, b_shape):
         return False
 
 
+def broadcastable(a_shape, b_shape, a_block_shape, b_block_shape):
+    if a_shape == b_shape:
+        return a_block_shape == b_block_shape
+
+    result_shape = broadcast_shape(a_shape, b_shape)
+    if result_shape is None:
+        return False
+    if len(a_shape) == len(b_shape):
+        return a_block_shape == b_block_shape
+    if a_shape == a_block_shape == () or b_shape == b_block_shape == ():
```

Comment:
This may be an ok fix, but I don't quite understand the original issue. Was it that shapes were `()` and block shapes were `(23,)`, or was the block shape for one of the arrays `()`?